packages/react-console/src/components/SerialConsole/XTerm.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+
+import { Terminal } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit';
+
+import { debounce } from '@patternfly/react-core';
+
+export interface XTermProps {
+  /** The number of columns to resize to */
+  cols?: number;
+  /** The number of rows to resize to */
+  rows?: number;
+  fontFamily?: string;
+  fontSize?: number;
+  /** Terminal title has been changed. */
+  onTitleChanged?: (title: string) => void;
+  /** Data to be sent from terminal to backend; (data) => {} */
+  onData?: (e: string) => void;
+  /** Terminal has been resized; backend shall be informed. (rows; cols) => {} */
+  onResize?: (rows: number, cols: number) => void;
+}
+
+export class XTerm extends React.Component<XTermProps> {
```

Comment:
FunctionComponent

Reply:
So this and the SerialConsole have two public method which should be accessed by the caller using a ref. In functional components refs dont work simply like that. I will need to use a forwardRef I think.
I will try to finish this today according to your comments.